src/update-changelog.ts

```diff
@@ -264,28 +289,31 @@ export async function updateChangelog({
       changelog.addRelease({ version: currentVersion });
     }
 
-    if (hasUnreleasedChanges) {
+    const hasUnreleasedChangesToRelease =
+      getKnownPropertyNames(changelog.getUnreleasedChanges()).length > 0;
+    if (hasUnreleasedChangesToRelease) {
       changelog.migrateUnreleasedChangesToRelease(currentVersion);
     }
+  }
 
-    const newChangeEntries = newCommits.map(({ prNumber, description }) => {
-      if (prNumber) {
-        const suffix = `([#${prNumber}](${repoUrl}/pull/${prNumber}))`;
-        return `${description} ${suffix}`;
-      }
-      return description;
-    });
+  const newChangeEntries = await getNewChangeEntries({
+    mostRecentTag,
+    repoUrl,
+    loggedPrNumbers: getAllLoggedPrNumbers(changelog),
+    projectRootDirectory,
+  });
 
-    for (const description of newChangeEntries.reverse()) {
-      changelog.addChange({
-        version: isReleaseCandidate ? currentVersion : undefined,
-        category: ChangeCategory.Uncategorized,
-        description,
-      });
-    }
+  for (const description of newChangeEntries.reverse()) {
+    changelog.addChange({
+      version: isReleaseCandidate ? currentVersion : undefined,
+      category: ChangeCategory.Uncategorized,
+      description,
+    });
   }
 
-  return changelog.toString();
+  const newChangelogContent = changelog.toString();
+  const isChangelogUpdated = changelogContent !== newChangelogContent;
+  return isChangelogUpdated ? newChangelogContent : undefined;
```

Comment:
`update()` in `cli.ts` expects that `updateChangelog` return `undefined` if there are no updates to be added to the changelog.